01-basics/20-weather/WeatherApp.js

```diff
@@ -3,49 +3,61 @@ import { getWeatherData, WeatherConditionIcons } from './weather.service.ts'
 
 export default defineComponent({
   name: 'WeatherApp',
+  setup() {
+    const weatherData = getWeatherData()
+    const weatherConditionIcons = WeatherConditionIcons;
 
-  template: `
-    <div>
-      <h1 class="title">Погода в Средиземье</h1>
+    const isDayTime = (current, sunrise, sunset) => {
+      return new Date(`1970-01-01T${current}:00`).getTime() > new Date(`1970-01-01T${sunrise}:00`).getTime() && new Date(`1970-01-01T${current}:00`).getTime() < new Date(`1970-01-01T${sunset}:00`).getTime();
+    }
 
-      <ul class="weather-list unstyled-list">
-        <li class="weather-card weather-card--night">
-          <div class="weather-alert">
+    return {
+      weatherData, weatherConditionIcons, isDayTime
+    }
+  },
+
+  template:
+    `<div v-for="(weather, index) in weatherData" key="weather.geographic_name">
+      <h1 v-if="index === 0" class="title">Погода в Средиземье</h1>
+
+      <ul class="weather-list">
+        <li class="weather-card" :class="!isDayTime(weather.current.dt, weather.current.sunrise, weather.current.sunset) ? 'weather-card--night' : ''">
+          <div v-if="weather.alert" class="weather-alert">
```

Comment:
This `v-for` makes a loop with a `div`. Each `div` has a list `ul` with only one list item `li` and one card. As the result, we there are many `ul` lists with only one list item in each of them. This doesn't make sense. `v-for` should be on `<li>` to render one list item for each weather item.

Reply:
Hmm, I supposed them like separate cards but, probable, you are right. Fixed.